code/src/components/TaskInput.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import styled from 'styled-components'
+import { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { tasks } from 'reducers/tasks'
+
+export const TaskInput = () => {
+
+  const [newTask, setNewTask] = useState('')
+  const dispatch = useDispatch()
+  const handleAddTask = () => {
+    dispatch(tasks.actions.addTask(newTask))
+    setNewTask('')
+  }
+
+  return (
+    <Wrapper>
+      <AddButton disabled={newTask.length === 0} onClick={handleAddTask}>+</AddButton>
+      <Input
+        type='text'
+        required
+        placeholder='Add task'
+        maxLength='100'
+        onChange={(event) => setNewTask(event.target.value)}
+        value={newTask}
+        // To add the task when hitting enter key
+        onKeyPress={(event) => { event.key === "Enter" && handleAddTask() }}

```

Comment:
Nice solution!

If you wrapped the input in a `<form>` tag and put the `handleAddTask` function as the `onSubmit` function of the form instead, then this would be the default behavior and you wouldn't have to add this logic. Also, you'd be able to remove the `onClick` of your button (and give it `type="submit"` instead) and then everything would submit the form, which would then invoke the `handleAddTask` function.